src/item/create/mutations.ts

```diff
@@ -0,0 +1,153 @@
+import {
+  DiscriminatedItem,
+  MAX_FILE_SIZE,
+  MAX_NUMBER_OF_FILES_UPLOAD,
+  partitionArray,
+} from '@graasp/sdk';
+import { FAILURE_MESSAGES, SUCCESS_MESSAGES } from '@graasp/translations';
+
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { AxiosProgressEvent } from 'axios';
+
+import { throwIfArrayContainsErrorOrReturn } from '../../api/axios.js';
+import { getKeyForParentId, itemsWithGeolocationKeys } from '../../keys.js';
+import { QueryClientConfig } from '../../types.js';
+import {
+  PostItemPayloadType,
+  PostItemWithThumbnailPayloadType,
+} from '../api.js';
+import { createItemRoutine, uploadFilesRoutine } from '../routines.js';
+import { postItem, postItemWithThumbnail, uploadFiles } from './api.js';
+
+export const usePostItem = (queryConfig: QueryClientConfig) => () => {
+  const queryClient = useQueryClient();
+  const { notifier } = queryConfig;
+  return useMutation(
+    async (item: PostItemPayloadType | PostItemWithThumbnailPayloadType) => {
+      // check if thumbnail was provided and if it is defined
+      if ('thumbnail' in item && item.thumbnail) {
+        return postItemWithThumbnail(item, queryConfig);
+      }
+      return postItem(item, queryConfig);
+    },
+    //  we cannot optimistically add an item because we need its id
+    {
+      onSuccess: () => {
+        notifier?.({
+          type: createItemRoutine.SUCCESS,
+          payload: { message: SUCCESS_MESSAGES.CREATE_ITEM },
+        });
+      },
+      onError: (error: Error) => {
+        notifier?.({ type: createItemRoutine.FAILURE, payload: { error } });
+      },
+      onSettled: (_data, _error, { geolocation, parentId }) => {
+        const key = getKeyForParentId(parentId);
+        queryClient.invalidateQueries(key);
+
+        // if item has geolocation, invalidate map related keys
+        if (geolocation) {
+          queryClient.invalidateQueries(itemsWithGeolocationKeys.allBounds);
+        }
+      },
+    },
+  );
+};
+
+/**
+ * @deprecated use useUploadFiles
+ * this mutation is used for its callback and invalidate the keys
+ * @param {UUID} id parent item id where the file is uploaded in
+ * @param {error} [error] error ocurred during the file uploading
+ */
+export const useUploadFilesFeedback =
```

Comment:
Will we be able to remove this one ?

Reply:
I think it's quite safe to remove it, I've set it to deprecated and we can remove at some point.